### PR TITLE
First release of the MSPI API

### PR DESCRIPTION
**Pull request by swift-tk**
In response to https://github.com/zephyrproject-rtos/zephyr/issues/75933

### CODE REVIEW

File: drivers/mspi/Kconfig [hunk 1]
```diff
@@ -8,6 +8,7 @@
 #
 menuconfig MSPI
 	bool "Multi-bit Serial Peripheral Interface (MSPI) bus drivers"
+	select EXPERIMENTAL
```
Comment (aescolar) [inline drivers/mspi/Kconfig:11]: This being a code change, maybe it should be in a different PR.

Comment (swift-tk) [reply drivers/mspi/Kconfig:11]: I was going to leave it out. Is marking the version to be 0.1.0 enough to mark API as experimental?
I’m not sure what is the impact of adding select EXPERIMENTAL as none of mspi related code uses that anyway.

Comment (aescolar) [reply drivers/mspi/Kconfig:11]: AFAIK they are a not exactly the same thing. The 0.1.0 in the header just affects the docs, so it will be visible as experimental in the API overview page.
The EXPERIMENTAL kconfig would show in the build .config, and if the user enabled CONFIG_WARN_EXPERIMENTAL trigger a warning during the kconfig build step.

Comment (aescolar) [reply drivers/mspi/Kconfig:11]: I think it will be best if you could split the kconfig change in a separate PR and commit. That way we can merge the rest as is for 3.7.0.
The kconfig change we may merge also for 3.7.0, or instead it may go to the 3.7 branch and be in 3.7.1

File: doc/releases/release-notes-3.7.rst [hunk 2]
```diff
@@ -745,6 +745,20 @@ Drivers and Sensors
   * Added release API
   * Added support for mode selection via the device tree
 
+* MSPI
+
+  * Add the new :ref:`MSPI(Multi-bit SPI) <mspi_api>` API, enabling support for advanced
```
Comment (aescolar) [inline doc/releases/release-notes-3.7.rst:750]:
```suggestion
  * Add the new experimental :ref:`MSPI(Multi-bit SPI) <mspi_api>` API, enabling support for advanced
```
maybe better?

File: doc/releases/release-notes-3.7.rst [hunk 3]
```diff
@@ -745,6 +745,20 @@ Drivers and Sensors
...
+    SPI controllers and peripherals that typically require command, address and data phases.
+    The API now supports single wire SDR up to hex wires DDR communication in sync/async ways.
```
Comment (aescolar) [inline doc/releases/release-notes-3.7.rst:752]:
```suggestion
    The API now supports from single wire SDR up to hex wires DDR communication in sync/async ways.
```
?

File: doc/releases/release-notes-3.7.rst [hunk 4]
```diff
@@ -745,6 +745,20 @@ Drivers and Sensors
...
+  * Added APS6404L QPI pSRAM device driver that uses the MSPI API.
```
Comment (aescolar) [inline doc/releases/release-notes-3.7.rst:753]:
```suggestion
  * Added APS6404L QPI pSRAM device driver.
```
Nit: I would probably just remove all 3 "that uses the MSPI API" from these 3 lines. It is implied from being in the list.

File: doc/releases/release-notes-3.7.rst [hunk 5]
```diff
@@ -745,6 +745,20 @@ Drivers and Sensors
...
+  * Added ATXP032 OPI NOR flash device driver that uses the MSPI API.
+  * Added Ambiq Apollo3 MSPI controller driver that implements the MSPI API.
+  * Added MSPI bus emulator under bus emulators to showcase the implementation of the MSPI API.
+  * Added MSPI flash device emulator to showcase the use of the MSPI API and interfacing with
+    MSPI bus controllers.
+  * Added mspi-async and mspi-flash example to showcase the use of MSPI device drivers.
```
Comment (aescolar) [inline doc/releases/release-notes-3.7.rst:759]:
```suggestion
  * Added :zephyr:code-sample:`mspi-async` and :zephyr:code-sample:`mspi-flash` samples to showcase the use of MSPI device drivers.
```
maybe better with links to their docs?

**Review by aescolar — CHANGES_REQUESTED** [review]
Thanks @swift-tk
Just a few minor requests.

File: drivers/mspi/Kconfig (hunk 1 above)
Comment (nashif) [reply drivers/mspi/Kconfig:11]: we should not need that, it experimental based on the API documentation below. We do not use this Kconfig for drivers that have a defined life cycle and a way to speficy stability.

Comment (swift-tk) [reply drivers/mspi/Kconfig]: Thanks and removed.

File: doc/releases/release-notes-3.7.rst (hunk 2 above)
Comment (swift-tk) [reply doc/releases/release-notes-3.7.rst]: Sure.

**Comment by swift-tk** [comment]
> Thanks @swift-tk Just a few minor requests.

Changes made, please take another look.

**Review by nashif — APPROVED** [review]

**Review by aescolar — APPROVED** [review]
